examples/cvode/parallel/cvAdvDiff_diag_p.c

```diff
@@ -45,6 +45,7 @@
 #include <cvode/cvode_diag.h>             /* prototypes for CVODE diagonal solver */
 #include <nvector/nvector_parallel.h>     /* access to MPI-parallel N_Vector     */
 #include <sundials/sundials_types.h>      /* definition of type realtype         */
+#include <sundials/sundials_mpi_errors.h>
```

Comment:
Small stylistic item:  I'm surprised that `sundials/sundials_mpi_errors.h` is #included before `mpi.h`, particularly since `sundials_mpi_errors.h` directly references `MPI_SUCCESS`.  I suppose that it works here because `nvector_parallel.h` itself includes MPI, but if we're already assuming this ordering then why does this routine directly include `mpi.h` at all?

Reply:
I think it is best the include whatever header files you directly use in the code in most cases. The exception being header files which are meant to bundle a bunch of includes together for easier including. In this case, I think `mpi.h` should be included since MPI is referenced in the example code directly, but you're right it does not need to be included since `nvector_parallel.h` includes it. How the include statements should be sorted is less clear to me (other than when the order matters to compile). Generally, I like std headers and TPL headers to go before the sundials ones. Following that, then `mpi.h` could be moved before the others but it would just be a stylistic choice.

Reply:
Also, following the include-what-you-use principle, a user should not have to worry about including mpi.h so that `sundials_mpi_errors.h` works. `sundials_mpi_errors.h` should include what it needs. So with that in mind, if `sundials_mpi_errors.h` doesnt do that already, it should.

Reply:
I've streamlined the includes now.